common/db/openssl/openssl.go

```diff
@@ -15,9 +16,10 @@ import (
 
 // For connecting to the database over ssl
 type SSLDBConnector struct {
-	dialInfo  *mgo.DialInfo
-	dialError error
-	ctx       *openssl.Ctx
+	dialInfo       *mgo.DialInfo
+	dialErrorMutex sync.Mutex
+	dialError      error
```

Comment:
Another option might be to log the openssl error when it happens instead of storing it in `dialError`. That makes the output very nosiy though:
```
$ ./bin/mongodump --host 983da27c-3f81-4414-8ea7-9322382bf222/localhost:27017,localhost:27018,localhost:27019 --db test --ssl
2017-05-23T15:22:12.241-0700	error dialing localhost:27018: SSL errors: SSL routines:ssl3_get_server_certificate:certificate verify failed
2017-05-23T15:22:12.241-0700	error dialing localhost:27019: SSL errors: SSL routines:ssl3_get_server_certificate:certificate verify failed
2017-05-23T15:22:12.241-0700	error dialing localhost:27017: SSL errors: SSL routines:ssl3_get_server_certificate:certificate verify failed
2017-05-23T15:22:12.749-0700	error dialing localhost:27017: SSL errors: SSL routines:ssl3_get_server_certificate:certificate verify failed
2017-05-23T15:22:12.750-0700	error dialing localhost:27018: SSL errors: SSL routines:ssl3_get_server_certificate:certificate verify failed
2017-05-23T15:22:12.750-0700	error dialing localhost:27019: SSL errors: SSL routines:ssl3_get_server_certificate:certificate verify failed
2017-05-23T15:22:13.256-0700	error dialing localhost:27018: SSL errors: SSL routines:ssl3_get_server_certificate:certificate verify failed
2017-05-23T15:22:13.256-0700	error dialing localhost:27017: SSL errors: SSL routines:ssl3_get_server_certificate:certificate verify failed
2017-05-23T15:22:13.257-0700	error dialing localhost:27019: SSL errors: SSL routines:ssl3_get_server_certificate:certificate verify failed
2017-05-23T15:22:14.265-0700	error dialing localhost:27017: SSL errors: SSL routines:ssl3_get_server_certificate:certificate verify failed
2017-05-23T15:22:14.266-0700	error dialing localhost:27019: SSL errors: SSL routines:ssl3_get_server_certificate:certificate verify failed
2017-05-23T15:22:14.266-0700	error dialing localhost:27018: SSL errors: SSL routines:ssl3_get_server_certificate:certificate verify failed
2017-05-23T15:22:14.773-0700	error dialing localhost:27019: SSL errors: SSL routines:ssl3_get_server_certificate:certificate verify failed
2017-05-23T15:22:14.773-0700	error dialing localhost:27018: SSL errors: SSL routines:ssl3_get_server_certificate:certificate verify failed
2017-05-23T15:22:14.774-0700	error dialing localhost:27017: SSL errors: SSL routines:ssl3_get_server_certificate:certificate verify failed
2017-05-23T15:22:15.277-0700	error dialing localhost:27018: SSL errors: SSL routines:ssl3_get_server_certificate:certificate verify failed
2017-05-23T15:22:15.278-0700	error dialing localhost:27017: SSL errors: SSL routines:ssl3_get_server_certificate:certificate verify failed
2017-05-23T15:22:15.278-0700	error dialing localhost:27019: SSL errors: SSL routines:ssl3_get_server_certificate:certificate verify failed
2017-05-23T15:22:15.783-0700	Failed: error connecting to db server: no reachable servers
```